### PR TITLE
pythonPackages.grpcio-tools: fix undefined symbol error on armv6l

### PR DESCRIPTION
**Pull request by lopsided98**
###### Motivation for this change

On armv6l, grpc needs to link to libatomic to provide C++ atomic operations. Without this PR, importing this Python module results in the following error:
```
Traceback (most recent call last):
  File "nix_run_setup", line 8, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
  File "setup.py", line 6, in <module>
    from grpc_tools import protoc
  File "/nix/store/b21rflbwjpfj7lxm3fg88p64ri6p0pm6-python3.7-grpcio-tools-1.23.0/lib/python3.7/site-packages/grpc_tools/protoc.py", line 20, in <module>
    from grpc_tools import _protoc_compiler
ImportError: /nix/store/b21rflbwjpfj7lxm3fg88p64ri6p0pm6-python3.7-grpcio-tools-1.23.0/lib/python3.7/site-packages/grpc_tools/_protoc_compiler.cpython-37m-arm-linux-gnueabihf.so: undefined symbol: __atomic_fetch_add_8
```

One might expect that the `grpcio` package would need this fix, rather than `grpcio-tools`, but this does not seem to be the case.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @vanschelven

### CODE REVIEW

File: pkgs/development/python-modules/grpcio-tools/default.nix [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{ stdenv, buildPythonPackage, fetchPypi, protobuf, grpcio }:
+{ stdenv, lib, buildPythonPackage, fetchPypi, protobuf, grpcio }:
```
Comment (FRidh) [inline pkgs/development/python-modules/grpcio-tools/default.nix]: this is an unnecessary change, please don't do that.

**Review by FRidh — CHANGES_REQUESTED** [review]

**Review by vanschelven — APPROVED** [review]
Looks good to me.

(Unchecked stuff ahead): looking at that stacktrace it almost looks like a bundled version of some protobuf implementation is what's doing the tripping-up here... which makes me wonder: how is the general state of affairs for the various protobuf-related libraries on armv6l ?

If the answer turns out to be "they don't work either", fixing them might be a separate PR.

**Comment by lopsided98** [comment]
I fixed protobuf on armv6l upstream a while ago (https://github.com/protocolbuffers/protobuf/pull/5227), and I have only seen this issue with `grpcio-tools`. I looked into fixing this upstream, but I think it would be difficult because it is hard to do autotools/CMake style compile tests in Python libraries.

**Comment by lopsided98** [comment]
It looks like this has been fixed upstream: https://github.com/grpc/grpc/pull/20414

**Comment by lopsided98** [comment]
Fixed by #72892
